### DISCUSSION TITLE
IPF map of SchmidFactor for complex stress tensor

### DISCUSSION THREAD

**Discussion by taknalms** [mtex-toolbox/mtex#2094] [Ask Anything]
Hello, 
I'm Alain, a PhD student in mechanics of materials and a new MTEX user.

First of all, thank you for the MTEX code, it's very useful. My question concerns a Schmid factor contour plot on an IPF. I calculate these factors with a stress tensor, in the tensile case I get what I want, however when the tensor changes the contour plot looks weird. 

I'd like to have IPFs for different slip systems with a torsional/shear (complex) stress
tensor as shown in the following image

![Capture d'écran 2024-03-18 174852](https://github.com/mtex-toolbox/mtex/assets/163892193/b9bdd0c3-9835-4f90-9913-de1828168e2c)

Best regards

PS: I'm working on a BCT material and my code is shown below
```matlab
sT =[1 0 0 ; 0 0 0 ;  0 0 0]; 

CS = crystalSymmetry('tetragonal',[5.83,5.83,3.18],'B-Sn');

ori = orientation.rand(6000,CS);

n = Miller(1,0,0,CS,'hkl');
d = Miller(0,1,0,CS,'uvw'); 
sS = slipSystem(d,n); 
sSAll = sS.symmetrise('antipodal'); 
sSAll_r = ori * sSAll;
schmidT = SchmidTensor(sSAll_r);

sF_ori = zeros(size(sSAll_r,2),6000);

for i = 1:size(sSAll);
    sF_ori(i,:) = abs(EinsteinSum(schmidT(:,i),[-1,-2],sT,[-1,-2])); 
    
end
vect = vector3d(1,0,0)
sF_max = max(sF_ori);

plotIPDF(ori,sF_max,vect,'contourf')`

```

**Comment by Tijmenvermeij** [comment]
Hi,

It would be helpful if you provide us with the figure Mtex gives you and an explanation on why you think the results are strange?

Best,
Tijmen
